lib/pdc/http/request/put_token.rb

```diff
@@ -0,0 +1,16 @@
+module PDC
+  class Request::PutTokenToHeader < Faraday::Middleware
+    include PDC::Logging
+
+    Faraday::Request.register_middleware :put_token_to_header => self
+
+    def call(env)
+      logger.debug "\n..... put token to header ..........................................".green
+      logger.debug self.class
+      logger.debug env.request_headers
+
+      env.request_headers['Token'] = PDC.token
```

Comment:
This depends on PDC class which imho points to tight coupling. I also think that PDC.token is a useful method as well. 
May I suggest moving the Token fetching part to a Module which both this middleware 
and PDC. includes?
That way the Dependencies would look like this

```
   [TokenFetcher]
          ^
          |
         / \
    [PDC]   [TokenMiddleware]
```